### ISSUE TITLE
Adding right context in Zipformer training and decoding

### ISSUE THREAD

**Issue by uni-bidisha** [k2-fsa/icefall#1514]
Hello, I am trying to add some frames of right context into the Zipformer training and decoding. I can see the training scripts of Zipformer uses different set of functions (forward passes) than the streaming_decode.py (streaming forward passes). Can you throw some idea on how to add right context in both training and decoding?

My another doubt is why attn_mask in not used for simulating streaming decoding to attend the left context chunks as in the following lines of code from zipformer.py

 ```
       if torch.jit.is_scripting() or torch.jit.is_tracing():
            # Not support exporting a model for simulating streaming decoding
            attn_mask = None
        else:
            attn_mask = self._get_attn_mask(x, chunk_size, left_context_chunks)
```

**Comment by danpovey** [comment]
The model itself can shift the labels a little, so adding explicit right context doesn't buy you much.
If we added right context to the individual layers it would make things more complicated.  Also the amount of
right context that is "seen" would compound over the layers, unless we did complicated things like recomputation of intermediate activations.  So to keep it simple, we don't use right context.

RE the attn_mask: when you are exporting an onnx model it's generally for production where you intend to use all context available.  I don't think a mask is even supported in the streaming_decode version of decoding.
